crates/wiggle/generate/src/types/variant.rs

```diff
@@ -96,6 +96,25 @@ pub(super) fn define_variant(names: &Names, name: &witx::Id, v: &witx::Variant)
         quote!()
     };
 
+    let enum_into = if v.cases.iter().all(|c| c.tref.is_none()) {
+        let into_repr_cases = v.cases.iter().enumerate().map(|(i, c)| {
+            let variant_name = names.enum_variant(&c.name);
+            let n = Literal::usize_unsuffixed(i);
+            quote!(#ident::#variant_name => #n)
+        });
+        quote! {
+            impl Into<#tag_ty> for #ident {
```

Comment:
Typically this is done with `From<#ident> for #tag_ty` instead of `Into` which automatically gives the `Into` trait implementation as well.